.agents/skills/wpilog-decode/scripts/find_wpilog_window.py

```
@@ -0,0 +1,158 @@
+    roots = sorted((Path.home() / "wpilib").glob("*"), reverse=True)
+    for root in roots:
+        if (root / "jdk").is_dir() and (root / "maven").is_dir():
+            return root
+    raise SystemExit("Could not find WPILib install under ~/wpilib")
+
+
+def wpilib_version(root: Path, group_path: str, artifact: str) -> str:
+    base = root / "maven" / group_path / artifact
+    versions = sorted([p.name for p in base.iterdir() if p.is_dir()])
+    if not versions:
+        raise SystemExit(f"No versions found for {artifact} under {base}")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)